### ISSUE TITLE
Will there be problem of using MobX4?

### ISSUE THREAD

**Issue by Albert-Gao** [mobxjs/mobx-react-lite#18]
`warning " > mobx-react-lite@0.2.0" has incorrect peer dependency "mobx@^5.0.0".`

I have to make compatible with IE11.
Is that fine?

**Comment by ilovedesert001** [comment]
it works fine to me .  mobx v4

**Comment by FredyC** [comment]
Well, as you can see I've made PR just to run tests against MobX 4 and some cases are failing. If you don't care about those cases, you are obviously free to use it. I haven't investigated it fully, might be some false positive, not sure.

I am not going to downgrade requirements for the package at this point as it should more about supporting bleeding edge.

**Comment by Albert-Gao** [comment]
I wonder why some tests would fail while `mobx-react-lite` use `React` new API underneath, and mobx v4 and v5 claims to have an API level compatiblity. And `React` doesn't drop support for IE 11.

So I cloned the repo and tried the tests with both v4 and v5.

Found something interesting, take the following test for example:

```javascript
        // observer.test.js
        test("inner store changed", () => {
            const { store, getAllByText, renderings } = execute()
            store.todos[0].title += "a"
           
            // line 80
            expect(getDNode(store, "todos").observers.size).toBe(1)
        })
```

This line fails:

```javascript
expect(getDNode(store, "todos").observers.size).toBe(1)
```

It's because v4 and v5 seem to have different implementation for this observable array, where it's `array` in v4 but `Set` in v5.

If you modified the line to:

```javascript
 expect(getDNode(store, "todos").value.length).toBe(1)
```

Then it will pass.

The original test seems to test the implementation detail? If we test against the unified API, then some tests should pass? But I could be wrong because I don't know much of this code base. Maybe we just want to test whether the inner observable has been initialized right and function well.

**Comment by FredyC** [comment]
@Albert-Gao Thanks for investigating it. I mostly copied tests from `mobx-react` an adapted a little bit (removed use of classes).

https://github.com/mobxjs/mobx-react/blob/master/test/observer.test.js#L81

It works there exactly the same and because mobx is v5 in devDependencies. 

So yea, ultimately there shouldn't be any problem using this module with mobx v4 as well.

**Comment by Albert-Gao** [comment]
Yes, I see.

But the peer dependency for `mobx-react` is been declared as

```json
"peerDependencies": {
    "mobx": "^4.0.0 || ^5.0.0",
    "react": "^0.13.0 || ^0.14.0 || ^15.0.0 || ^16.0.0"
  },
```

Do you think that we should update the `peerDependencies` for `mobx-react-lite`? :)

**Comment by FredyC** [comment]
I don't see a reason. As you have investigated it works with v4 just fine, only some internal stuff is different which is not used by the package itself.

**Comment by Albert-Gao** [comment]
Just try to make people clear that it's compatible with both version. Otherwise when installing dependencies, there will be warnings like `peerDependency not match`, forgot that exact words, but you get the idea.

**Comment by FredyC** [comment]
Ah, I see now. I thought we do have v4 in peer dependencies. Good point :)

Fixed in ec271b2 and released v0.3.1

**Comment by Albert-Gao** [comment]
@FredyC Wonderful! Thanks!
